homeassistant/components/imap/__init__.py

```diff
@@ -188,6 +196,42 @@ async def async_delete(call: ServiceCall) -> None:
 
     hass.services.async_register(DOMAIN, "delete", async_delete, SERVICE_DELETE_SCHEMA)
 
+    async def async_fetch(call: ServiceCall) -> ServiceResponse:
+        """Process mark as seen service call."""
+        entry_id: str = call.data[CONF_ENTRY]
+        uid: str = call.data[CONF_UID]
+        _LOGGER.debug(
+            "Fetch text for message %s. Entry: %s",
+            uid,
+            entry_id,
+        )
+        client = await async_get_imap_client(hass, entry_id)
+        try:
+            response = await client.fetch(uid, "BODY.PEEK[]")
+        except (TimeoutError, AioImapException) as exc:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="imap_server_fail",
+                translation_placeholders={"error": str(exc)},
+            ) from exc
+        raise_on_error(response, "fetch_failed")
+        message = ImapMessage(response.lines[1])
+        await client.close()
+        return {
```

Comment:
We could also put flags similar to https://github.com/home-assistant/core/pull/115108 in the service call and return only the selected items.

Reply:
My aim is only to return text and some basic items as the rest can be retrieved from the event info. Further the info needs to be serializable.